presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -884,6 +884,26 @@ public static Slice stEndPoint(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return serialize(createFromEsriGeometry(lines.getPoint(lines.getPointCount() - 1), reference));
     }
 
+    @SqlNullable
+    @Description("Returns an array of points in a LINESTRING geometry")
+    @ScalarFunction("ST_Points")
+    @SqlType("array(" + GEOMETRY_TYPE_NAME + ")")
+    public static Block stPoints(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        validateType("ST_Points", geometry, EnumSet.of(LINE_STRING));
+        if (geometry.isEmpty()) {
+            return null;
+        }
+        MultiPath lines = (MultiPath) geometry.getEsriGeometry();
+        SpatialReference reference = geometry.getEsriSpatialReference();
```

Comment:
Spatial reference is always null, hence, drop this variable is use `null` below.

Reply:
@cooldoger That's correct. If you'd like to remove spatial references from other places, do so in a separate commit.

Reply:
Here is a a separate PR for that: #12309